### PR TITLE
client: add UidCopyWithData to expose UIDPLUS COPYUID

### PR DESCRIPTION
**Pull request by madirey**
## Description

Adds `Client.UidCopyWithData`, which performs a `UID COPY` and returns the UIDPLUS `COPYUID` data (RFC 4315) — the UIDs the copied messages were assigned in the destination mailbox. Today `UidCopy` discards the command's status response, so destination UIDs are unrecoverable except by searching the destination folder afterward, which is fragile: it requires a Message-ID or a complete header set and can match zero or many messages.

Motivation: in go-mantis, IMAP move-to-spam (and other move-based remediations) recovers a moved message's new UID by searching the destination folder by header. When no Message-ID / complete fallback header is available, that search fails and a move that actually succeeded gets reported as a failure and retried. `COPYUID` returns the new UIDs directly from the copy, eliminating the search on UIDPLUS servers (and should also be covered on non-UIDPLUS servers that are following the spec).

## How did you validate this change
Added unit tests, researched and matched edge cases and response codes as best as I can determine. Nothing is using this code until it's implemented in `go-mantis`.

- Assisted by Claude 🤖

### CODE REVIEW

File: client/cmd_selected.go [hunk 1]
```diff
@@ -277,6 +279,82 @@ func (c *Client) UidCopy(seqset *imap.SeqSet, dest string) error {
 	return c.copy(true, seqset, dest)
 }
 
+// CopyData holds the COPYUID data from the UIDPLUS extension (RFC 4315),
+// reporting the UIDs assigned to the copied messages in the destination
+// mailbox. DestUIDs corresponds positionally to SourceUIDs.
```
[the remark this answers is not in the document]
Comment (madirey) [reply client/cmd_selected.go]: Good catch — fixed in 44adfc5. Dropped the positional-correspondence claim: although RFC 4315 defines the source/dest sets as positionally corresponding on the wire, `ParseSeqSet` sorts and coalesces ranges, so that pairing isn't recoverable from the returned `*imap.SeqSet`. The doc now just describes the two sets, and notes that for a single-message copy each set holds one UID (so `DestUIDs` is unambiguous — which is the use case this is for).

File: client/cmd_selected.go [hunk 2]
```diff
@@ -277,6 +279,82 @@ func (c *Client) UidCopy(seqset *imap.SeqSet, dest string) error {
...
+type CopyData struct {
+	UIDValidity uint32
+	SourceUIDs  *imap.SeqSet
+	DestUIDs    *imap.SeqSet
+}
+
+// UidCopyWithData is identical to UidCopy, but additionally returns the COPYUID
+// data (RFC 4315) when the server advertises UIDPLUS. This lets callers learn
```
[the remark this answers is not in the document]
Comment (madirey) [reply client/cmd_selected.go]: Fixed in 44adfc5. Reworded to describe the actual trigger — the presence of a `COPYUID` response code (which UIDPLUS-capable servers send on a successful copy) — rather than implying a capability check, since the implementation keys off `status.Code`, not `CAPABILITY`.

File: client/cmd_selected.go [hunk 3]
```diff
@@ -277,6 +280,170 @@ func (c *Client) UidCopy(seqset *imap.SeqSet, dest string) error {
...
+// memory.
+func parseUIDList(set string) ([]uint32, error) {
+	if set == "" {
+		return nil, fmt.Errorf("empty uid set")
+	}
+
+	var uids []uint32
+	for _, part := range strings.Split(set, ",") {
+		lo, hi, isRange := strings.Cut(part, ":")
+
```
[the remark this answers is not in the document]
Comment (madirey) [reply client/cmd_selected.go:395]: Fixed in 990a854

**Review by alexk307 — APPROVED** [review]
Nice!
